common/persistence/cassandra/metadata_store.go

```diff
@@ -165,19 +177,54 @@ func (m *MetadataStore) CreateNamespaceInV2Table(
 		return nil, serviceerror.NewUnavailable(fmt.Sprintf("CreateNamespace operation failed. Inserting into namespaces table. Error: %v", err))
 	}
 	defer func() { _ = iter.Close() }()
-
-	if !applied {
-		// Namespace already exist.  Delete orphan namespace record before returning back to user
+	deleteOrphanNamespace := func() {
+		// Delete namespace from `namespaces_by_id`
 		if errDelete := m.session.Query(templateDeleteNamespaceQuery, request.ID).WithContext(ctx).Exec(); errDelete != nil {
 			m.logger.Warn("Unable to delete orphan namespace record. Error", tag.Error(errDelete))
 		}
+	}
 
-		if id, ok := previous["Id"].([]byte); ok {
-			msg := fmt.Sprintf("Namespace already exists.  NamespaceId: %v", primitives.UUIDString(id))
+	if !applied {
+
+		// if both conditions fail, find the one related to the first query
+		matched, err := fieldMatches(previous, "name", request.Name)
+		if err != nil {
+			return nil, err
+		}
+		if !matched {
+			m := make(map[string]interface{})
+			if iter.MapScan(m) {
+				if applied, ok := m["[applied]"].(bool); ok && !applied {
```

Comment:
will additional rows from `iter` have a different value of `applied` than the first row? If no, then looks like in the `!matched` case, we should always set `previous` to the row from `iter`, if it exists.